test/11_holograph_factory_tests.ts

```diff
@@ -1,61 +1,242 @@
 
-  describe(`bridgeIn()`, async function () {
-    it('should return the expected selector from the input payload');
-    it('should return bad data if payload data is invalid');
+    chainId = (await ethers.provider.getNetwork()).chainId;
+
+    Mock = await ethers.getContractFactory('Mock');
+    mock = await Mock.deploy();
+    await mock.deployed();
+
+    mockSigner = await ethers.getSigner(mock.address);
   });
 
-  describe('bridgeOut()', async function () {
-    it('should return selector and payload');
+  describe('init():', async () => {
+    // TODO: Check initialized
+    // it.only('should check that contract was successfully initialized once', async () => {
+    //   await expect(l1.holographFactory.connect(deployer)._isInitialized().to.equal(true);
+    // });
+
+    it('should fail if already initialized', async () => {
+      const initCode = generateInitCode(
+        ['address', 'address'],
+        [l1.holographFactory.address, l1.holographRegistry.address]
+      );
+      await expect(l1.holographFactory.connect(deployer).init(initCode)).to.be.reverted;
+    });
   });
 
-  describe('deployHolographableContract()', async function () {
-    it('should fail with invalid signature if config is incorrect');
-    it('should fail with invalid signature if signature.r is incorrect');
-    it('should fail with invalid signature if signature.s is incorrect');
-    it('should fail with invalid signature if signature.v is incorrect');
-    it('should fail with invalid signature if signer is incorrect');
+  describe('deployHolographableContract()', async () => {
+    it('should fail with invalid signature if config is incorrect', async () => {
+      let { erc20Config, erc20ConfigHash, erc20ConfigHashBytes } = await generateErc20Config(
+        l1.network,
+        l1.deployer.address,
+        'hToken',
+        l1.network.tokenName + ' (Holographed #' + l1.network.holographId.toString() + ')',
+        'h' + l1.network.tokenSymbol,
+        l1.network.tokenName + ' (Holographed #' + l1.network.holographId.toString() + ')',
+        '1',
+        18,
+        ConfigureEvents([]),
+        generateInitCode(['address', 'uint16'], [l1.deployer.address, 0]),
+        l1.salt
+      );
+
+      let hTokenErc20Address = await l1.registry.getHolographedHashAddress(erc20ConfigHash);
+      hTokenErc20Address = await l1.registry.getHolographedHashAddress(erc20ConfigHash);
+
+      let sig = await l1.deployer.signMessage(erc20ConfigHashBytes);
+      let signature: Signature = StrictECDSA({
+        r: '0x' + sig.substring(2, 66),
+        s: '0x' + sig.substring(66, 130),
+        v: '0x' + sig.substring(130, 132),
+      } as Signature);
+
+      await expect(
+        l1.holographFactory.connect(deployer).deployHolographableContract(erc20Config, signature, owner.address)
+      ).to.be.revertedWith('HOLOGRAPH: invalid signature');
+    });
+
+    // TODO: Not sure how to test if each value of signature is correct
```

Comment:
Yea i may have gone overboard about this. I suppose if i could override the internal values after you create the signature.